src/Utilities/Extensions/IOperationExtensions.cs

```diff
@@ -301,5 +301,112 @@ public static IAnonymousObjectCreationOperation GetAnonymousObjectCreation(this
 
         public static bool IsInsideCatchClause(this IOperation operation)
             => operation.GetAncestor<ICatchClauseOperation>(OperationKind.CatchClause) != null;
+
+        public static bool HasAnyOperationDescendant(this ImmutableArray<IOperation> operationBlocks, Func<IOperation, bool> predicate)
+        {
+            foreach (var operationBlock in operationBlocks)
+            {
+                if (operationBlock.HasAnyOperationDescendant(predicate))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public static bool HasAnyOperationDescendant(this IOperation operationBlock, Func<IOperation, bool> predicate)
+        {
+            Debug.Assert(operationBlock != null);
+            Debug.Assert(predicate != null);
+            foreach (var descendant in operationBlock.DescendantsAndSelf())
+            {
+                if (predicate(descendant))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public static bool HasAnyOperationDescendant<TOperation>(this ImmutableArray<IOperation> operationBlocks)
```

Comment:
Rather than casting, this should be using OperationKind. Integer checks will be much more performant than casting.